### PR TITLE
CO-3910 Add "needs consent" state for W&P sponsorships

### PR DESCRIPTION
**Pull request by TheRaphael0000**

### CODE REVIEW

File: partner_compassion/models/partner_compassion.py [hunk 1]
```diff
@@ -255,9 +257,46 @@ class ResPartner(models.Model):
...
+        [
+            ("not_submitted", _("Not submitted yet.")),
+            ("waiting", _("Waiting Compassion approval")),
+            ("approved", _("Approved")),
+            ("refused", _("Refused")),
+        ],
+        string="Parent consents",
+        default="not_submitted",
+        store=True,
+        readonly=False,
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]:
```suggestion
```

File: partner_compassion/models/partner_compassion.py [hunk 2]
```diff
@@ -255,9 +257,46 @@ class ResPartner(models.Model):
...
+        required=True,
+        track_visibility="onchange",
+    )
+
+    can_manage_paid_sponsorships = fields.Boolean(
+        compute="_compute_can_manage_paid_sponsorships",
+        store=False,
+        readonly=True,
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]:
```suggestion
        help="Sponsor has 18 years old or has parents consent for paying sponsorship"
```

File: partner_compassion/models/partner_compassion.py [hunk 3]
```diff
@@ -255,9 +257,46 @@ class ResPartner(models.Model):
...
+    )
+
     ##########################################################################
     #                             FIELDS METHODS                             #
     ##########################################################################
+
+    @api.multi
+    def _compute_is_of_age(self):
+        for record in self:
+            record.is_of_age = record.age >= self.MAJORITY_AGE
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]: I don't see anywhere the field `age` existing in the `res.partner` model, or did I miss it?

File: partner_compassion/models/partner_compassion.py [hunk 4]
```diff
@@ -255,9 +257,46 @@ class ResPartner(models.Model):
...
+        required=True,
+        track_visibility="onchange",
+    )
+
+    can_manage_paid_sponsorships = fields.Boolean(
+        compute="_compute_can_manage_paid_sponsorships",
+        store=False,
+        readonly=True,
+    )
+    is_of_age = fields.Boolean(
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]: `is_adult` or `has_majority` would be a better field name.

File: partner_compassion/views/partner_compassion_view.xml [hunk 5]
```diff
@@ -92,6 +92,11 @@
                 <field name="user_login" attrs="{'invisible': [('user_login', '=', False)]}"/>
                 <field name="birthdate_date" attrs="{'invisible': [('is_company','=',True)]}"/>
                 <field name="deathdate" attrs="{'invisible': [('is_company','=',True)]}"/>
+
+                <field name="is_of_age" invisible="1"/>
+                <field name="age"/>
+                <field name="parent_consent" attrs="{'invisible': [('is_of_age','=',True)]}"/>
+                <field name="can_manage_paid_sponsorships" attrs="{'invisible': [('is_of_age','=',True)]}"/>
```
Comment (ecino) [inline partner_compassion/views/partner_compassion_view.xml]: I think this field can only stay internal to the code logic and doesn't need to be seen in the view.
```suggestion
```

**Review by ecino — CHANGES_REQUESTED** [review]

File: partner_compassion/models/partner_compassion.py (hunk 3 above)
Comment (TheRaphael0000) [reply partner_compassion/models/partner_compassion.py]: It's in additional addons, partner_contact_birthdate

**Pull request merged**